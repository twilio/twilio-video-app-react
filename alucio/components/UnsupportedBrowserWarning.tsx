import React from 'react';
import Video from 'twilio-video';

export default function({ children }: { children: React.ReactElement }) {
  if (!Video.isSupported) {
    return (
      <div>
        Not Supported Browser
      </div>
    );
  }

  return children;
}
