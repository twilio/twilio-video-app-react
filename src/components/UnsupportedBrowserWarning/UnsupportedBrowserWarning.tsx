import React from 'react';
import Video from 'twilio-video';
import { Container, Link, Typography, Paper, Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  container: {
    marginTop: '2.5em',
  },
  paper: {
    padding: '1em',
  },
  heading: {
    marginBottom: '0.4em',
  },
});

export default function({ children }: { children: React.ReactElement }) {
  const classes = useStyles();

  if (!Video.isSupported) {
    return (
      <Container>
        <Grid container justify="center" className={classes.container}>
          <Grid item xs={12} sm={6}>
            <Paper className={classes.paper}>
              <Typography variant="h4" className={classes.heading}>
                Browser or context not supported
              </Typography>
              <Typography>
                Please open this application in one of the{' '}
                <Link
                  href="https://www.twilio.com/docs/video/javascript#supported-browsers"
                  target="_blank"
                  rel="noopener"
                >
                  supported browsers
                </Link>
                .
                <br />
                If you are using a supported browser, please ensure that this app is served over a{' '}
                <Link
                  href="https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts"
                  target="_blank"
                  rel="noopener"
                >
                  secure context
                </Link>{' '}
                (e.g. https or localhost).
              </Typography>
            </Paper>
          </Grid>
        </Grid>
      </Container>
    );
  }

  return children;
}
