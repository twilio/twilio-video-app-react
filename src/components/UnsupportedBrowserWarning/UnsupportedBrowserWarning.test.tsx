import React from 'react';
import UnsupportedBrowserWarning from './UnsupportedBrowserWarning';
import Video from 'twilio-video';
import { shallow } from 'enzyme';

describe('the UnsupportedBrowserWarning component', () => {
  it('should render correctly when isSupported is false', () => {
    // @ts-ignore
    Video.isSupported = false;
    const wrapper = shallow(
      <UnsupportedBrowserWarning>
        <span>Is supported</span>
      </UnsupportedBrowserWarning>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children when isSupported is true', () => {
    // @ts-ignore
    Video.isSupported = true;
    const wrapper = shallow(
      <UnsupportedBrowserWarning>
        <span>Is supported</span>
      </UnsupportedBrowserWarning>
    );
    expect(wrapper.text()).toBe('Is supported');
  });
});
