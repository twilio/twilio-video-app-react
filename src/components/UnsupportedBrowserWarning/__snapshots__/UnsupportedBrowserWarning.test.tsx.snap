// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UnsupportedBrowserWarning component should render correctly when isSupported is false 1`] = `
<WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-container-1"
    container={true}
    justifyContent="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-2"
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-heading-3"
          variant="h4"
        >
          Browser or context not supported
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          Please open this application in one of the
           
          <WithStyles(ForwardRef(Link))
            href="https://www.twilio.com/docs/video/javascript#supported-browsers"
            rel="noopener"
            target="_blank"
          >
            supported browsers
          </WithStyles(ForwardRef(Link))>
          .
          <br />
          If you are using a supported browser, please ensure that this app is served over a
           
          <WithStyles(ForwardRef(Link))
            href="https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts"
            rel="noopener"
            target="_blank"
          >
            secure context
          </WithStyles(ForwardRef(Link))>
           
          (e.g. https or localhost).
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
